custom_components/petsafe/SensorEntities.py

```diff
@@ -233,5 +234,32 @@ async def async_update(self) -> None:
             self._attr_native_value = datetime.datetime.fromtimestamp(
                 feeding["payload"]["time"], pytz.timezone("UTC")
             )
-
+        if self._device_type == "next_feeding":
+            data: PetSafeData = self.coordinator.data
+            feeder: petsafe.devices.DeviceSmartFeed = next(
+                x for x in data.feeders if x.api_name == self._api_name
+            )
+            schedules = await feeder.get_schedules()
```

Comment:
I think this implementation is going to result in excessive API calls since it's going to poll at the same frequency as the rest of the polling (every 10s). This doesn't seem like something we need to refresh every 10 seconds. I think we need to think through a more efficient implementation.

Reply:
Totally fair. I'll put this in draft and dig a bit deeper.

Reply:
Basically my concern is we may end up hitting 429s. Could we consider something like, don't refresh again until the current value of `next_feeding is < now`?

Reply:
ah, yes, that's simple enough :-) Thanks